### PR TITLE
API-17394 Part 7 Eventing breakup - refactor hlr & sc controller specs

### PR DESCRIPTION
**Pull request by ElleeB**
Part 7 updates:

HLR and SC controller specs to use with settings necessary now that job are called directly from #update_status

### CODE REVIEW

File: modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb [hunk 1]
```diff
@@ -143,12 +143,20 @@ def new_base_path(path)
       allow(client_stub).to receive(:upload).and_return(faraday_response)
       allow(faraday_response).to receive(:success?).and_return(true)
 
-      Sidekiq::Testing.inline! do
-        post(path, params: @data, headers: @headers)
-      end
+      with_settings(Settings.vanotify.services.lighthouse.template_id,
```
Comment (cilestin) [inline modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb:146]: This test is misnamed (and apparently has been for a while). The test says that it "create[s] the job to build the PDF" but this is not testing that, it's testing email notifications. Let's update the test description to be more accurate to what's being tested.

File: modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb [hunk 2]
```diff
@@ -253,12 +253,20 @@ def obj.class
       allow(client_stub).to receive(:upload).and_return(faraday_response)
       allow(faraday_response).to receive(:success?).and_return(true)
 
-      Sidekiq::Testing.inline! do
-        post(path, params: data, headers: headers)
-      end
+      with_settings(Settings.vanotify.services.lighthouse.template_id,
```
Comment (cilestin) [inline modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb:256]: Same here. There's nothing that's testing the creation of the PDF job.

**Review by cilestin — CHANGES_REQUESTED** [review]

File: modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb (hunk 1 above)
Comment (ElleeB) [reply modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb:146]: on it

File: modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb (hunk 2 above)
Comment (ElleeB) [reply modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb:256]: on it

**Review by cilestin — APPROVED** [review]

**Pull request merged**